src/main/java/uk/gov/pay/api/resources/directdebit/DirectDebitPaymentsResource.java

```diff
@@ -67,4 +80,43 @@ public Response createPayment(@ApiParam(value = "accountId", hidden = true) @Aut
         return Response.created(publicApiUriGenerator.getDirectDebitPaymentURI(payment.getPaymentId()))
                 .entity(payment).build();
     }
+
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "OK", response = DirectDebitSearchResponse.class),
+            @ApiResponse(code = 401, message = "Credentials are required to access this resource"),
+            @ApiResponse(code = 422, message = "Invalid parameter: from_date. See Public API documentation for the correct data formats", response = PaymentError.class),
+            @ApiResponse(code = 429, message = "Too many requests", response = ApiErrorResponse.class),
+            @ApiResponse(code = 500, message = "Downstream system error", response = PaymentError.class)})
+    public Response searchPayments(@ApiParam(value = "accountId", hidden = true)
+                                   @Auth Account account,
+                                   @ApiParam(value = "Your payment reference to search", hidden = false)
+                                   @QueryParam("reference") String reference,
+                                   @ApiParam(value = "State of payments to be searched. Example=success", hidden = false, allowableValues = "pending,success,failed,cancelled,expired")
```

Comment:
Where are these values coming from? From https://github.com/alphagov/pay-publicapi/blob/master/src/main/java/uk/gov/pay/api/model/directdebit/mandates/MandateStatus.java: created, started, pending, submitted, active, inactive, cancelled

Reply:
https://github.com/alphagov/pay-direct-debit-connector/blob/a1345682f860e7005d76e5e2d984f30ec9e60838/src/main/java/uk/gov/pay/directdebit/payments/api/ExternalPaymentState.java

But "started" isn't actually a thing so I've left that out.

Reply:
ah ok. i didn't realise statuses are being got from ExternalPaymentState.